frontend/packages/knative-plugin/src/components/overview/EventPubSubSubscribers.tsx

```diff
@@ -0,0 +1,102 @@
+    subscribers?: Subscriber[];
+  };
+};
+
+export const PubSubSubscribers: React.FC<PubSubSubscribersProps> = ({ item }) => {
+  const [expanded, setExpanded] = React.useState('');
+  const [isExpanded, setIsExpanded] = React.useState(false);
+
+  const ExpandButton = (isFilterExpanded, onClick) => {
+    const title = `${isExpanded ? 'Hide' : 'Show'} filters`;
+    return (
+      <Button
+        aria-label={title}
+        className="subscriber-list__expand-button"
+        onClick={onClick}
+        title={title}
+        variant="plain"
+      >
+        {title}
+        {isFilterExpanded ? (
+          <AngleDownIcon className="subscriber-list__expand-icon" />
+        ) : (
+          <AngleRightIcon className="subscriber-list__expand-icon" />
+        )}
+      </Button>
+    );
+  };
+
+  return (
+    <>
+      <SidebarSectionHeading text="Subscribers" />
+      {item?.subscribers.length > 0 ? (
+        <ul className="list-group subscriber-list">
+          {item?.subscribers.map((sub) => (
+            <li className="list-group-item" key={sub.name}>
+              <ResourceLink
+                kind={referenceFor(sub)}
+                name={sub?.name}
+                namespace={item.obj.metadata.namespace}
+              />
+              {sub?.data.length > 0 &&
+                sub?.data.map((r) => {
+                  const onClick = () => {
+                    if ((isExpanded && expanded !== r?.name) || expanded.length === 0) {
+                      setExpanded(r?.name);
+                      setIsExpanded(true);
+                    } else {
+                      setExpanded('');
+                      setIsExpanded(false);
+                    }
+                  };
+
+                  return (
+                    <React.Fragment key={r?.name}>
+                      <Grid className="subscriber-list__relationship">
+                        <GridItem span={9}>
+                          <ResourceLink
+                            className=""
+                            kind={referenceFor(r)}
+                            name={r?.name}
+                            namespace={item.obj.metadata.namespace}
+                          />
+                        </GridItem>
+                        <GridItem span={3}>
+                          {r?.filters && r?.filters.length > 0 && (
+                            <span className="text-right">
+                              {ExpandButton(r?.name === expanded && isExpanded, onClick)}
+                            </span>
+                          )}
+                        </GridItem>
+                      </Grid>
+                      {r?.filters && expanded === r.name && (
+                        <Grid className="row subscriber-list__relationship-table">
+                          <GridItem>
+                            <FilterTable filters={r?.filters} />
+                          </GridItem>
+                        </Grid>
+                      )}
```

Comment:
I'm fine with this, But we can probably get away with using the GridItem with span={12} in the above Grid itself for FilterTable. I tried this locally and seemed to work fine.
<img width="531" alt="Screenshot 2020-07-29 at 11 30 58 AM" src="https://user-images.githubusercontent.com/9278015/88762819-8c253a80-d18f-11ea-97ad-786b8fbb0bf8.png">

Reply:
@sahil143  I haven't added the span, becuase as per design, we dont need any padding around the filter table. 
eg: 
![image](https://user-images.githubusercontent.com/9964343/88767969-ff32af00-d197-11ea-99aa-2866e7fe671b.png)

Reply:
Ah! okay, make sense. You missed the color of `hide filters` text.